raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -1436,6 +1438,28 @@ public void testInvalidFetchRequest() throws Exception {
         context.assertSentFetchPartitionResponse(Errors.INVALID_REQUEST, epoch, OptionalInt.of(localId));
     }
 
+    @Test
+    public void testFetchRequestVersionHandling() throws Exception {
+        int localId = 0;
+        int otherNodeId = 1;
+        int epoch = 5;
+        Set<Integer> voters = Utils.mkSet(localId, otherNodeId);
+
+        RaftClientTestContext context = RaftClientTestContext.initializeAsLeader(localId, voters, epoch);
+
+        // Latest fetch request.
+        FetchRequestData fetchRequestData = context.fetchRequest(
+                epoch, context.clusterId.toString(), otherNodeId, 0L, 0, 0);
+        context.deliverRequest(fetchRequestData);
+        context.pollUntilResponse();
+        context.assertSentFetchPartitionResponse(Errors.NONE, epoch, OptionalInt.of(localId));
+
+        FetchRequest.maybeDownGradeReplicaState(fetchRequestData, (short) 14);
```

Comment:
nit: Could we add a comment here to be consistent with the previous block?

Reply:
Refactored.